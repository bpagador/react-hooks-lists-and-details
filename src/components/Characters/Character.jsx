import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, image }) => {
  return (
    <section>
      <h4>{name}</h4>
      <img src={image} alt={name} width="250" />
    </section>
  );
};

Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};
