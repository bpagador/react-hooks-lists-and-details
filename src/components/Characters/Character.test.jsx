import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';


describe('single Character', () => {
  it('returns a single character', () => {
    const wrapper = shallow(
      <Character name="Shelley" image="https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
