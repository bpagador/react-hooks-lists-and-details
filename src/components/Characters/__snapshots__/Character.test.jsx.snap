// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single Character returns a single character 1`] = `
<section>
  <h4>
    Shelley
  </h4>
  <img
    alt="Shelley"
    src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
    width="250"
  />
</section>
`;
